### ISSUE TITLE
If audio muted in VLC, then mpv will have no sound, and can't un-mute within mpv

### ISSUE THREAD

**Issue by DoomsDay007** [mpv-player/mpv#14479]
### mpv Information

```bash
mpv v0.38.0-dirty Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on May 23 2024 10:30:57
libplacebo version: v6.338.2
FFmpeg version: n7.0.1
FFmpeg library versions:
   libavutil       59.8.100
   libavcodec      61.3.100
   libavformat     61.1.100
   libswscale      8.1.100
   libavfilter     10.1.100
   libswresample   5.1.100
```

### Other Information

- Linux version: Arch Linux
- Kernel Version: 6.9.7-arch1-1-surface
- GPU Model: VGA compatible controller - Intel - Iris Graphics 540 [8086:1926] (rev 0a)
- Mesa/GPU Driver Version: Mesa 24.1.2-arch1.1
- Window Manager and Version: KWin 6.1.1.2-1 (KDE 6.1.1)
- Source mpv: Arch extra repo
- Introduced in version: unknown

### Reproduction Steps

Have both mpv and VLC installed

Open a video file in VLC

Mute the video in VLC, i.e. hit _m_ to use the keyboard shortcut for mute

Quit VLC without un-muting audio, i.e. hit _Ctrl+q_ to use the keyboard shortcut for quit

Open any video file in mpv (does not matter if it is the same one opened in VLC or not)

mpv plays the video file but there is no sound

### Expected Behavior

mpv plays video files with sound, or alternatively, mpv should be able to un-mute the sound

### Actual Behavior

mpv plays video files but there is no sound, and it cannot un-mute the sound

Note:
I first noticed this issue in jellyfin-media-player because it uses mpv as its backend, and when trying to play any video in my library it did not have any sound.

While writing this issue report, I also noticed that if I dug a little in the system sound settings I could un-mute the sound for mpv (outside of mpv), however, I could not find a way to un-mute the sound from within mpv. mpv's mute and un-mute appears to work independently of whatever system VLC uses to mute the audio.
Also note that, other sounds on the system were working just fine the whole time, i.e. YouTube videos played fine with sound, and sound was heard when using the system's speaker test. So it isn't like the whole system was muted, just certain programs.

### Log File

[output.txt](https://github.com/user-attachments/files/16075130/output.txt)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by DoomsDay007** [comment]
I'll add that I'm not 100% sure if I should add this issue here or to VLC's github, but I just thought I should let you know because the issue affects mpv (whether directly or indirectly).

Also, thought it might be helpful to anyone that comes across this issue, because I was just trying to watch a movie and I could not get the sound to work, and it was a pain in the butt to figure out what was going on at first lol.

**Comment by CounterPillow** [comment]
You need to actually show mpv exhibiting the behaviour in your log file, not just printing its help output.

One reason I can think of what's happening is that you are using alsa, and VLC is muting audio at the mixer. Even if your mpv is using pipewire or pulseaudio, it would then be muted, since another application helpfully muted the entire audio device.

mpv's mute is not muting the system's audio output device, as that would be stupid. In general, it will not mess with the system mixer at all. It can control the audio output device's mute state with the `ao-mute` property, but I don't recommend it.

**Comment by DoomsDay007** [comment]
Okay, no problem. I'll try posting the log file again. Hopefully, I do it right this time lol. I'm not a noob, but I'm not a programmer either, just a somewhat tech-savvy user trying to be helpful. :)

[output.txt](https://github.com/user-attachments/files/16074933/output.txt)

**Comment by DoomsDay007** [comment]
@CounterPillow 

> One reason I can think of what's happening is that you are using alsa, and VLC is muting audio at the mixer. Even if your mpv is using pipewire or pulseaudio, it would then be muted, since another application helpfully muted the entire audio device.

Yeah, perhaps that could be it. Yes, that would be ever so "helpful" wouldn't it. xD
The weird part that really threw me off is, like I said, other sounds on the system were working just fine the whole time. YouTube videos played fine with sound, and sound was heard when using the system's speaker test. So it isn't like the whole system was muted, just certain programs. 

The way I came across the issue in the first place was kinda random anyway. Yesterday, I was watching a video file that was saved on my computer in VLC. Then I muted it because my wife came in and started talking to me. Later on I closed VLC and started working on something else. Today I was trying to trying to play a video from my Jellyfin server using jellyfin-media-player (with the mpv backend) and I couldn't get any sound. I tried downloading the video from my server and playing it locally in mpv and VLC (which was still muted) and it didn't have any sound. It was only after I realized that VLC still was muted and un-muted it that the sound started working again everywhere.

> mpv's mute is not muting the system's audio output device, as that would be stupid. 

Agreed. I don't know if that is what's going on here, but agreed.

> In general, it will not mess with the system mixer at all. It can control the audio output device's mute state with the ao-mute property, but I don't recommend it.

Okay, gotcha. Good to know. Thank you.

**Comment by t-8ch** [comment]
> It can control the audio output device's mute state with the ao-mute property, but I don't recommend it.

At least for `ao_pulse` and `ao_pipewire`, `ao-mute` does not touch the device but only mpvs private sink node.
So other applications are not impacted.

**Comment by zsnero** [comment]
Adding `ao=pulse` or `ao=alsa` on top of mpv.conf file fixes the issue. You can test this by running `mpv --ao=pulse video.mp4`

**Comment by cek333** [comment]
Thanks DoomsDay007 for filing this report. This behaviour has been most puzzling.

For me, I had `audio-device=pulse` in my mpv.conf file. Commenting out that line was enough to remove vlc's influence on mpv.

**Comment by llyyr** [comment]
This should be fixed by https://github.com/mpv-player/mpv/pull/17035 already, are you not using 0.41?

**Comment by cek333** [comment]
> This should be fixed by [#17035](https://github.com/mpv-player/mpv/pull/17035) already, are you not using 0.41?

```
mpv --version
mpv v0.41.0+git20260309.07c3ff3725 Copyright © 2000-2026 mpv/MPlayer/mplayer2 projects
libplacebo version: v7.360.1
FFmpeg version: 8.1.2
FFmpeg library versions:
   libavcodec      62.28.102
   libavdevice     62.3.102
   libavfilter     11.14.102
   libavformat     62.12.102
   libavutil       60.26.102
   libswresample   6.3.102
   libswscale      9.5.102
```

**Comment by pinkflames** [comment]
It might be worth opening alsamixer and checking what the mixer state is when you mute and unmute audio in VLC. If VLC is directly controlling the mixer state, that will be visible there. Remember that F6 key in alsamixer allows changing the mixer shown and it might be worth checking both the default mixer (which should be either the pulse or pipewire plugin for ALSA) and the actual audio adapter mixer.

Another thing to try would be setting `--pipewire-volume-mode=global` mpv option and see, if that makes ao_pipewire able to deal with this on its end.

The WP volumes should be stored in `~/.local/state/wireplumber/stream-properties` but, if VLC is directly controlling the ALSA mixer, then I don't think there's much reason to peer at that file. ;)

**Comment by cek333** [comment]
Muting/unmuting VLC doesn't appear to have any effect in alsamixer. This is true for both the default mixer (pipewire) and the actual audio adapter (a Realtek ALC3254). 

For the 2nd suggestion, I first restored the `audio-device=pulse` line in my `mpv.conf` file. A muted VLC still meant mpv was muted even after using this new option: `mpv --pipewire-volume-mode=global *.mp4`

**Comment by pinkflames** [comment]
I just tried this on my end and either `--pipewire-volume-mode=global` doesn't do what I thought it does or it's is plain busted. But that's just a side quest, let's not get bogged down by that.

If the actual ALSA mixer state is not being changed, then whatever is going on is probably doing so via `.local/state/wireplumber/stream-properties` . Since it might contain private information, don't just post it online but do look over it. Most likely VLC and mpv end up sharing an entry.

Something like `watch -d10 -n 1 grep -v media.name ~/.local/state/wireplumber/stream-properties` (which should filter out all the spam of per-file or per-youtube entries) might be worth a try but, since I can't reproduce this issue on my end, no promises it's actually gonna show anything useful. While watch is running, try adjusting volume in VLC and see if there's anything interesting printed by watch in reaction to VLC.

**Comment by cek333** [comment]
When I ran `watch -d10 -n 1 grep -v media.name ~/.local/state/wireplumber/stream-properties` and then toggled the mute in vlc, no additional entries get printed to the screen. In fact, there are no vlc related entries in `~/.local/state/wireplumber/stream-properties` at all.
So, `grep vlc  ~/.local/state/wireplumber/stream-properties` yields no output. Versus `grep mpv  ~/.local/state/wireplumber/stream-properties` which yields a number of hits.
